tools/tpm2_createprimary.c

```diff
@@ -293,24 +294,26 @@ bool tpm2_tool_onstart(tpm2_options **opts) {
     return *opts != NULL;
 }
 
-int tpm2_tool_onrun(TSS2_SYS_CONTEXT *sapi_context, tpm2_option_flags flags) {
+#define VALID_CTX(ctx) \
+	(ctx.flags.A && ctx.flags.g && ctx.flags.G)
 
+int tpm2_tool_onrun(TSS2_SYS_CONTEXT *sapi_context, tpm2_option_flags flags) {
     UNUSED(flags);
-    int returnVal = 0;
+    int returnVal;
 
-    if(ctx.flags.A == 1 && ctx.flags.g == 1 && ctx.flags.G == 1) {
-        returnVal = create_primary(sapi_context);
+    if (VALID_CTX(ctx) != true) {
+        LOG_ERR("Invalid option flags");
+        return -EINVAL;
+    }
 
-        if (returnVal == 0 && ctx.flags.C) {
-            returnVal = files_save_tpm_context_to_path(sapi_context, ctx.handle2048rsa,
-                                                       ctx.context_file) != true;
-        }
+    returnVal = create_primary(sapi_context);
+    if (returnVal != 0 || ctx.flags.C != 1) {
+        return returnVal;
+    }
 
-        if(returnVal) {
-            return 1;
-        }
-    } else {
-        return 1;
+    if (files_save_tpm_context_to_path(sapi_context, ctx.handle2048rsa,
+                                       ctx.context_file) != true) {
```

Comment:
```
bool res = files_save_tpm_context_to_path();
if (!res) {
    / /ERROR
}
```
But you could also go for the tail call here,

return !files_save_tpm_context_to_path();